### PR TITLE
[MERGE] base, event: "add properties" is now default, add properties to event

### PR DESCRIPTION
**Pull request by schoffeniels**
### 1. Move "Add Properties" button to cog menu by default
When the property field is added into a form view, the "add a property"
button is shown by default.
However, most of the time the users won't add properties when opening a
form view, making this button take up space for nothing.

To alleviate this issue, [this commit](https://github.com/odoo/odoo/commit/1f80810e39f811122a65e11cd8cac99aa39a7960) added an option that allows to hide
this button, and added actions to allow adding properties from the cog
menu instead.

Now that properties are being added into several views, one would like
this "add properties" button to be hidden by default, and show an entry
in the cogmenu allowing add properties by default, without needing to
create an action for every view using a property field.
Therefore, a new static action "Add Properties" has been added in form
controller that is available if a properties field has been encountered
during the parsing of the form view's arch.

The "hideAddButton" has been changed into "showAddButton" and now allows
to show the "Add a Property" by default.

### 2. Add properties (and related properties definitions) on the following models:
- analytic.account (analytic.plan)
- event.registration (event.event)
- hr.employee (hr.res_company)
- hr_recruitment.hr_applicant (hr_recruitment.hr_job)
- product.product (product.category)
- sale.sale_order (sale.crm_team)
- stock.picking (stock.picking_type)

Task-3458627

### CODE REVIEW

File: addons/hr/models/res_company.py [hunk 1]
```diff
@@ -9,3 +9,4 @@ class Company(models.Model):
 
     hr_presence_control_email_amount = fields.Integer(string="# emails to send")
     hr_presence_control_ip_list = fields.Char(string="Valid IP addresses")
+    employee_properties_definition = fields.PropertiesDefinition('Employee Properties')
```
Comment (awa-odoo) [inline addons/hr/models/res_company.py:12]: Do we really want to add more things onto the company?

Also, the commit message mentions that it depends on the "hr.company" model, but it's "res.company".

**Review by awa-odoo — COMMENTED** [review]
Hello @schoffeniels 

One minor comment, otherwise nothing much to add compared to the first validation.

Thanks & cheers!

**Comment by awa-odoo** [comment]
For reference, review was initially done here: https://github.com/odoo/odoo/pull/132578

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**